### PR TITLE
LTTng relayd Mac OSX support

### PR DESCRIPTION
**Pull request by mjeanson**
Build with:

./configure --disable-bin-lttng --disable-bin-lttng-consumerd --disable-bin-lttng-crash --disable-bin-lttng-sessiond --disable-extras --without-lttng-ust --disable-man-page

### CODE REVIEW

File: src/common/compat/fcntl.h [hunk 1]
```diff
@@ -78,7 +78,7 @@ static inline int posix_fadvise(int fd, off_t offset, off_t len, int advice)
 }
 #endif
 
-#if !(defined(__linux__) || defined(__FreeBSD__) || defined(__CYGWIN__) || defined(__sun__))
+#if !(defined(__linux__) || defined(__FreeBSD__) || defined(__CYGWIN__) || defined(__sun__) || defined(__APPLE__))
 #error "Please add support for your OS."
 #endif /* __linux__ , __FreeBSD__, __CYGWIN__, __sun__ */
```
Comment (PSRCode) [inline src/common/compat/fcntl.h]: Update comment for endif.

File: src/common/compat/time.h [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        tp->tv_sec = now.tv_sec;
+        tp->tv_nsec = now.tv_nsec;
+
+deallocate:
+	mach_port_deallocate(mach_task_self(), clock);
+end:
+	return ret;
+}
+
+#else
```
Comment (jgalar) [inline src/common/compat/time.h]: ```/* __APPLE__ */```

File: src/common/compat/time.h [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+static inline
+int lttng_clock_gettime(clockid_t clk_id, struct timespec *tp)
+{
+	return clock_gettime(clk_id, tp);
+}
+
+#endif
```
Comment (jgalar) [inline src/common/compat/time.h]: ```/* __APPLE__ */```

**Comment by jgalar** [comment]
Adding `time.h` to `src/common/compat/Makefile.am`.

**Comment by jgalar** [comment]
Merged, thanks!

**Pull request closed without merging**